src/dmd/root/speller.d

```diff
@@ -95,83 +95,82 @@ private auto spellerY(alias dg)(const(char)[] seed, size_t index, ref int cost)
     return p; // return "best" result
 }
 
-private auto spellerX(alias dg)(const(char)* seed, size_t seedlen, bool flag)
+private auto spellerX(alias dg)(const(char)[] seed, bool flag)
 {
-    if (!seedlen)
+    if (!seed.length)
         return null;
     char[30] tmp;
-    char* buf;
-    if (seedlen <= tmp.sizeof - 2)
-        buf = tmp.ptr;
+    char[] buf;
+    if (seed.length <= tmp.sizeof - 1)
+        buf = tmp;
     else
     {
-        buf = cast(char*)alloca(seedlen + 2); // leave space for extra char
-        if (!buf)
-            return null; // no matches
+        buf = (cast(char*)alloca(seed.length + 1))[0 .. seed.length + 1]; // leave space for extra char
     }
     int cost = int.max, ncost;
     searchFunctionType!dg p = null, np;
     /* Deletions */
-    buf[0 .. seedlen] = seed[1 .. seedlen + 1];
-    for (size_t i = 0; i < seedlen; i++)
+    buf[0 .. seed.length - 1] = seed[1 .. $];
```

Comment:
Why the -1?

Reply:
No longer need to copy a terminating null.